packages/amplify-cli/src/pre-deployment-pull.ts

```diff
@@ -2,31 +2,30 @@ import { createAmplifySkeletonProject } from 'amplify-app';
 import fetch from 'node-fetch';
 import path from 'path';
 import fs from 'fs-extra';
-import { spawn } from 'child_process';
+import { $TSContext } from 'amplify-cli-core';
 
-export async function preDeployPullBackend(context, sandboxId: string) {
+export async function preDeployPullBackend(context: $TSContext, sandboxId: string) {
   const url = `https://rh2kdo2x79.execute-api.us-east-1.amazonaws.com/gamma/AppState/${sandboxId}`;
 
   // Fetch schema
   const res = await fetch(`${url}`);
   const resJson = await res.json();
 
   // App not present
-  if (resJson.message && resJson.message === 'Requested app was not found') {
+  if (resJson.message === 'Requested app was not found') {
     context.print.error('Requested app was not found');
     return;
-    // throw new Error('Requested app was not found');
   }
 
   // Handle Deployed App Case
   if (resJson.appId) {
-    context.print.error('App is already deployed');
+    context.print.error(`This app is already deployed. You can pull it using "amplify pull --appId ${resJson.appId}"`);
```

Comment:
You'd also have to pass the env name for post pull deployment to work. Please add that too! :D

Reply:
If you don't specify an env name, it will prompt you to pick one